### ISSUE TITLE
[Bug]: Babel @babel/preset-env cannot find module 'regjsgen'

### ISSUE THREAD

**Issue by vincelawdev** [babel/babel#13309]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@rollup/plugin-babel

### Input code

rollup.config.js:

`import resolve from '@rollup/plugin-node-resolve';
import babel from '@rollup/plugin-babel';

export default {
    input: 'assets/source/scripts/main.js',
    output: {
        file: 'assets/build/scripts/main.js',
        format: 'iife',
    },
    plugins: [
        resolve(),
        babel({ 
            babelHelpers: 'bundled'
        })
    ]
};`

assets/source/scripts/main.js

Does not matter if this file is empty or has code. The error is always as follows.

`# yarn build
yarn run v1.22.10
$ rollup --config

assets/source/scripts/main.js → assets/build/scripts/main.js...
[!] (plugin babel) Error: [BABEL] C:\Local Sites\assets\source\scripts\main.js: Cannot find module 'regjsgen'
Require stack:
- C:\Local Sites\node_modules\regexpu-core\rewrite-pattern.js
- C:\Local Sites\node_modules\@babel\helper-create-regexp-features-plugin\lib\index.js
- C:\Local Sites\node_modules\@babel\plugin-transform-dotall-regex\lib\index.js
- C:\Local Sites\node_modules\@babel\preset-env\lib\available-plugins.js
- C:\Local Sites\node_modules\@babel\preset-env\lib\filter-items.js
- C:\Local Sites\node_modules\@babel\preset-env\lib\index.js
- C:\Local Sites\node_modules\@babel\core\lib\config\files\module-types.js
- C:\Local Sites\node_modules\@babel\core\lib\config\files\configuration.js
- C:\Local Sites\node_modules\@babel\core\lib\config\files\index.js
- C:\Local Sites\node_modules\@babel\core\lib\index.js
- C:\Local Sites\node_modules\@rollup\plugin-babel\dist\index.js
- C:\Local Sites\rollup.config.js
- C:\Local Sites\node_modules\rollup\dist\shared\loadConfigFile.js
- C:\Local Sites\node_modules\rollup\dist\bin\rollup (While processing: "C:\\Local Sites\\node_modules\\@babel\\preset-env\\lib\\index.js")`

=======

### Configuration file name

package.json

### Configuration

{
  "name": "whatever",
  "version": "0.0.3963",
  "description": "Theme",
  "engines": {
    "node": ">=10.0.0",
    "npm": ">=6.0.0"
  },
  "scripts": {
    "build": "rollup --config"
  },
  "dependencies": {
    "bootstrap-sass": "^3.4.1",
    "jquery": "^3.5.1"
  },
  "devDependencies": {
    "@babel/core": "7.14.2",
    "@babel/plugin-transform-classes": "^7.14.2",
    "@babel/preset-env": "7.14.2",
    "@rollup/plugin-babel": "^5.3.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "autoprefixer": "^10.2.5",
    "browserslist": "^4.16.6",
    "chalk": "^4.1.0",
    "del": "^6.0.0",
    "fancy-log": "^1.3.3",
    "graceful-fs": "^4.2.2",
    "gulp": "^4.0.2",
    "gulp-bump": "^3.2.0",
    "gulp-clean-css": "^4.2.0",
    "gulp-concat": "^2.5.2",
    "gulp-image": "^6.2.1",
    "gulp-plumber": "^1.2.1",
    "gulp-postcss": "^9.0.0",
    "gulp-real-favicon": "^0.3.2",
    "gulp-sass": "^4.0.2",
    "gulp-sourcemaps": "^3.0.0",
    "merge-stream": "^2.0.0",
    "postcss": "^8.2.15",
    "rollup": "^2.47.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": [
            "last 2 versions",
            "not ie <= 10"
          ]
        },
        "useBuiltIns": "entry",
        "debug": true,
        "exclude": "node_modules/**"
      }
    ],
    "plugins": [
      "@babel/plugin-transform-classes"
    ]
  }
}

### Current and expected behavior

I am trying to retire my Gulp tasks and move them over to Rollup.

I am having issues with @babel/preset-env. If I include it as a preset with or without any option config, I get an error. The babel plugins run fine without the @babel/preset-env added to preset.

I have no idea why I am getting a "Cannot find module 'regjsgen'" error because I do not use regjsgen anywhere in my code.

The expected behaviour is the code should transpile with @babel/preset-env preset.

### Environment

Yarn 1.22.10
Node 14.16.1
Babel 7.14.2
Rollup 2.47.0
Windows 10

### Additional context

Just for the record, the package.json config below works without @babel/preset-env added to the babel preset.

`{
  "name": "whatever",
  "version": "0.0.3963",
  "description": "Theme",
  "engines": {
    "node": ">=10.0.0",
    "npm": ">=6.0.0"
  },
  "scripts": {
    "build": "rollup --config"
  },
  "dependencies": {
    "bootstrap-sass": "^3.4.1",
    "jquery": "^3.5.1"
  },
  "devDependencies": {
    "@babel/core": "7.14.2",
    "@babel/plugin-transform-classes": "^7.14.2",
    "@babel/preset-env": "7.14.2",
    "@rollup/plugin-babel": "^5.3.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "autoprefixer": "^10.2.5",
    "browserslist": "^4.16.6",
    "chalk": "^4.1.0",
    "del": "^6.0.0",
    "fancy-log": "^1.3.3",
    "graceful-fs": "^4.2.2",
    "gulp": "^4.0.2",
    "gulp-bump": "^3.2.0",
    "gulp-clean-css": "^4.2.0",
    "gulp-concat": "^2.5.2",
    "gulp-image": "^6.2.1",
    "gulp-plumber": "^1.2.1",
    "gulp-postcss": "^9.0.0",
    "gulp-real-favicon": "^0.3.2",
    "gulp-sass": "^4.0.2",
    "gulp-sourcemaps": "^3.0.0",
    "merge-stream": "^2.0.0",
    "postcss": "^8.2.15",
    "rollup": "^2.47.0"
  },
  "babel": {
    "plugins": [
      "@babel/plugin-transform-classes"
    ]
  }
}`

**Comment by vincelawdev** [comment]
I updated my package.json babel settings to the following. I used an array for the preset entry.

`"babel": {
    "presets": [
      [
        "@babel/preset-env",
        {
          "targets": {
            "browsers": [
              "last 2 versions",
              "not ie <= 10"
            ]
          },
          "modules": false,
          "useBuiltIns": "entry",
          "corejs": 3,
          "debug": true,
          "exclude": "node_modules/**"
        }
      ]
    ]
  },`

This is my rollup config at the moment.

`import eslint from '@rollup/plugin-eslint';
import { uglify } from 'rollup-plugin-uglify';
import { babel } from '@rollup/plugin-babel';

export default {
    input: 'assets/source/scripts/main.js',
    output: {
        file: 'assets/build/scripts/main.js',
        format: 'iife',
        sourcemap: true,
    },
    watch: {
        include: ['assets/source/scripts/**/*.js']
    },
    plugins: [
        eslint(),
        babel({ babelHelpers: 'bundled' }),
        uglify(),
    ]
};`

Getting a different error message now.

`assets/source/scripts/main.js → assets/build/scripts/main.js...
[!] (plugin babel) TypeError: [BABEL] C:\Local Sites\assets\source\scripts\main.js: plugins.map is not a function (While processing: "C:\\Local Sites\\node_modules\\@babel\\preset-env\\lib\\index.js")
assets\source\scripts\main.js
TypeError: [BABEL] C:\Local Sites\assets\source\scripts\main.js: plugins.map is not a function (While processing: "C:\\Local Sites\\node_modules\\@babel\\preset-env\\lib\\index.js")
    at expandIncludesAndExcludes (C:\Local Sites\node_modules\@babel\preset-env\lib\normalize-options.js:47:35)
    at normalizeOptions (C:\Local Sites\node_modules\@babel\preset-env\lib\normalize-options.js:128:19)
    at C:\Local Sites\node_modules\@babel\preset-env\lib\index.js:271:37
    at C:\Local Sites\node_modules\@babel\helper-plugin-utils\lib\index.js:22:12
    at async (C:\Local Sites\node_modules\@babel\core\lib\gensync-utils\async.js:43:33)
    at async (C:\Local Sites\node_modules\gensync\index.js:186:15)
    at C:\Local Sites\node_modules\gensync\index.js:216:13
    at Generator.next (<anonymous>)
    at C:\Local Sites\node_modules\@babel\core\lib\config\full.js:215:21
    at Generator.next (<anonymous>)`

**Comment by nicolo-ribaudo** [comment]
This looks like an installation problem. Can you try deleting `node_modules` and reinstalling it?

**Comment by vincelawdev** [comment]
Yes, I did try deleting node_modules and running yarn install again.

**Comment by vincelawdev** [comment]
I just tried to delete node_modules and ran yarn install. Same error as above. I am considering dropping Babel-preset-evn and just install a handful of Babel plugins.

**Comment by fedeci** [comment]
Have you tried also to regenerate the `yarn.lock` file?

**Comment by vincelawdev** [comment]
Yes, I have also tried delating my yarn.lock file when I deleted node_modules and ran yarn install again. No luck here.

**Comment by JLHwung** [comment]
What's the output of `npm ls regjsgen`? Here is what it normally looks like
```
project@version /path/to/project
├─┬ @babel/preset-env@7.13.15
│ └─┬ @babel/plugin-proposal-unicode-property-regex@7.12.13
│   └─┬ @babel/helper-create-regexp-features-plugin@7.12.17
│     └─┬ regexpu-core@4.7.1
│       └── regjsgen@0.5.1
```

If you can't find `regjsgen`, try to find `regexpu-core` and all the way up-to `@babel/preset-env`. This should help you pin point what is the exact package not downloaded due to installation issue.

**Comment by vincelawdev** [comment]
npm ls regjsgen

+-- @babel/preset-env@7.14.2
| `-- @babel/plugin-proposal-unicode-property-regex@7.12.13
|   `-- @babel/helper-create-regexp-features-plugin@7.12.17
|     `-- regexpu-core@4.7.1
|       `-- UNMET DEPENDENCY regjsgen@^0.5.1
`--  extraneous error: Failed to parse json

npm ERR! extraneous: regjsgen C:\Users\Vince\Local Sites\node_modules\regjsgen
npm ERR! error in C:\Users\Vince\Local Sites\node_modules\regjsgen: Failed to parse json
npm ERR! Unexpected token � in JSON at position 0 while parsing near '�o�s�t�i�c�i☻...'
npm ERR! missing: regjsgen@^0.5.1, required by regexpu-core@4.7.1

**Comment by vincelawdev** [comment]
npm ls regexpu-core

`-- @babel/preset-env@7.14.2
  `-- @babel/plugin-proposal-unicode-property-regex@7.12.13
    `-- @babel/helper-create-regexp-features-plugin@7.12.17
      `-- regexpu-core@4.7.1

**Comment by JLHwung** [comment]
> npm ERR! error in C:\Users\Vince\Local Sites\node_modules\regjsgen: Failed to parse json
npm ERR! Unexpected token � in JSON at position 0 while parsing near '�o�s�t�i�c�i☻...'

As you can see the package.json of `regjsgen` is corrupted, they should look like https://unpkg.com/regjsgen@0.5.2/package.json, you can try removing `node_modules/regjsgen` and re-install, or try to download the tarball shown in `npm view regjsgen dist.tarball` and extract for yourself to see what is going wrong here.

**Comment by vincelawdev** [comment]
When I was debugging this issue a few days ago, I noticed the package.json of regjsgen was empty.

I have no idea what might have caused this issue. Do we need to get the author of regjsgen to fix something or use an older version of regjsgen with this preset to make it work?

**Comment by nicolo-ribaudo** [comment]
It's possible that there is a cache between you and npm (maybe a company proxy?) and the `package.json` file is corrupted. However, there is nothing we can do on our side to fix it :shrug:

**Comment by vincelawdev** [comment]
I don't use a proxy at home. Anyway, I tested yarn install with and without my VPN software running. The output is still the same. When I open the regjsgen 0.5.2 package.json in my node_modules, VS Code warns that it is "binary or uses an unsupported text encoding".

**Comment by vincelawdev** [comment]
If I run `yarn add regjsgen@0.5.1 --dev`, I can open the package.json without any issues. I am guessing the issue is with regjsgen@0.5.2.

**Comment by vincelawdev** [comment]
| -- regexpu-core@4.7.1 | -- UNMET DEPENDENCY regjsgen@^0.5.1

Seems like yarn install could not install regjsgen@^0.5.1. It installed regjsgen@^0.5.2 instead.

**Comment by existentialism** [comment]
@vincelawdev fwiw, i have no issues with regjsgen@0.5.2 locally and it seems fine on https://unpkg.com/browse/regjsgen@0.5.2/package.json. I'll echo @nicolo-ribaudo and say it's really unlikely to be anything for us to fix.

**Comment by vincelawdev** [comment]
I found a fix and I do not quite understand why this is happening. I have tried to clear my npm cache, tried npm install instead of yarn install, tried updating the npm/yarn registry to https://registry.npmjs.org, reinstalled latest LTS version of Node.js, installed the latest version of npm/yarn, etc.

This was what worked for me in my package.json:

"devDependencies": {
"@babel/plugin-transform-literals": "7.12.1",
"regjsgen": "0.5.1",
},
"resolutions": {
    "@babel/plugin-transform-literals": "7.12.1",
    "regjsgen": "0.5.1"
  },

I forced Babel to use these two versions. @babel/plugin-transform-literals 7.12.13 and regjsgen 0.5.2 fails for me for unknown reasons. I had to drop one version down for these two packages.
